webapp/channels/src/components/suggestion/search_date_suggestion/search_date_suggestion.tsx

```diff
@@ -2,84 +2,72 @@
+    const handleDayClick = useCallback((day: Date) => {
         const dayString = new Date(Date.UTC(day.getFullYear(), day.getMonth(), day.getDate())).toISOString().split('T')[0];
-        this.props.onClick(dayString, this.props.matchedPretext);
-    };
-
-    handleKeyDown = (e: KeyboardEvent) => {
-        if (Keyboard.isKeyPressed(e, Constants.KeyCodes.DOWN) && document.activeElement?.id === 'searchBox') {
-            this.setState({datePickerFocused: true});
-        } else if (Keyboard.isKeyPressed(e, Constants.KeyCodes.ESCAPE)) {
-            this.props.handleEscape?.();
-        }
-    };
-
-    componentDidMount() {
-        document.addEventListener('keydown', this.handleKeyDown);
-    }
-
-    componentWillUnmount() {
-        document.removeEventListener('keydown', this.handleKeyDown);
-    }
-
-    iconLeft = () => {
+        onClick(dayString, matchedPretext);
+    }, [onClick, matchedPretext]);
+
+    useEffect(() => {
+        const handleKeyDown = (e: KeyboardEvent) => {
+            if (Keyboard.isKeyPressed(e, Constants.KeyCodes.DOWN) && document.activeElement?.id === 'searchBox') {
+                setDatePickerFocused(true);
+            } else if (Keyboard.isKeyPressed(e, Constants.KeyCodes.ESCAPE)) {
+                handleEscape?.();
+            }
+        };
+        document.addEventListener('keydown', handleKeyDown);
+        return () => {
+            document.removeEventListener('keydown', handleKeyDown);
+        };
+    }, []);
+
+    const iconLeft = () => {
         return (
             <i className='icon icon-chevron-left'/>
         );
     };
 
-    iconRight = () => {
+    const iconRight = () => {
         return (
             <i className='icon icon-chevron-right'/>
         );
     };
 
-    render() {
-        const locale: string = this.props.locale;
-
-        this.loadedLocales = Utils.getDatePickerLocalesForDateFns(locale, this.loadedLocales);
-
-        return (
-            <DayPicker
-                onDayClick={this.handleDayClick}
-                showOutsideDays={true}
-                mode={'single'}
-                locale={this.loadedLocales[locale]}
-                initialFocus={this.state.datePickerFocused}
-                onMonthChange={this.props.preventClose}
-                id='searchDatePicker'
-                selected={this.props.currentDate}
-                components={{
-                    IconRight: this.iconRight,
-                    IconLeft: this.iconLeft,
-                }}
-            />
-        );
-    }
-}
+    return (
+        <DayPicker
+            onDayClick={handleDayClick}
+            showOutsideDays={true}
+            mode={'single'}
+            locale={loadedLocales[locale]}
+            initialFocus={datePickerFocused}
+            onMonthChange={preventClose}
+            id='searchDatePicker'
+            selected={currentDate}
+            components={{
+                IconRight: iconRight,
+                IconLeft: iconLeft,
+            }}
```

Comment:
To avoid passing a new object on every render, and since we are moving `iconLeft` and `iconRight` out of the component, move this also out of the component into a constant.

Reply:
Done